### PR TITLE
Get register of interest via API

### PR DESCRIPTION
**Pull request by ajparsons**
Parliament's turned off their old site, so our scraper is broken!

This scopes out the old parser, adds a new parser that converts the Parliament API into the XML TheyWorkForYou expects.

As seen in `dailyupdate`, this can be run via:

```bash
poetry run python -m pyscraper.regmem download-all-registers --chamber commons
```

This updates some requirements in the poetry, and also *uses* this poetry config for the first time in production.

Rather than exactly replicating the previous structure, there's now a bit more use of lists to reflect how parliament sees items and subitems.

I've tested this in both the import to database:

image

And the comparison over time:

image

There's a boundary problem in that the sudden change will break the retrospect. If we delete and recreate the last few items though, this should be less of a problem. 

Also have the option if reviewing this will take a while to just upload the generated files manually (given they seem fine).

### CODE REVIEW

**Comment by ajparsons** [comment]
twfy host now has poetry - so this should work even in advance of the twfy poetry pr.

File: pyscraper/regmem/__main__.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+@cli.command()
+@click.option("--chamber", type=str, default="commons")
+@click.option("--date", type=datetime.date)
+@click.option("--force-refresh", is_flag=True)
+@click.option("--quiet", is_flag=True)
+def download_register_for_date(
+    chamber: str, date: datetime.date, force_refresh: bool = False, quiet: bool = False
+):
+    if chamber == "commons":
+        commons_process.download_register_from_date(
```
Comment (dracos) [inline pyscraper/regmem/__main__.py:35]: why does for_date become from_date :)

File: pyscraper/regmem/commons/api_models.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
+from __future__ import annotations
+
+from datetime import date
+from enum import Enum
+from typing import Any, List, Optional
+
+from jinja2 import Template
```
Comment (dracos) [inline pyscraper/regmem/commons/api_models.py:7]: jinja2 is not in the requirements

File: pyscraper/regmem/commons/process.py [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+
+        member.categories = categories
+
+        members.append(member)
+
+    members.sort(key=lambda x: x.id)
+
+    # given we pull in parent interests when not in the category, we need to check we haven't duplicated
+
+    existing_ids = []
```
Comment (dracos) [inline pyscraper/regmem/commons/process.py:162]: Should these be sets, if I'm understanding it right? Could get same ID more than once

**Review by dracos — CHANGES_REQUESTED** [review]
One issue with missing jinja2. To be honest, I didn't follow all the code (perhaps some doctests of e.g. move_subitems_under_parent to show an example would be nice, not sure I understood the "parents" stuff), but if you're happy with the output, fine with me :)

File: pyscraper/regmem/commons/process.py (hunk 3 above)
Comment (ajparsons) [reply pyscraper/regmem/commons/process.py:162]: Yeah, it's not big enough for a real performance penalty but makes it clearer they should be unique.

File: pyscraper/regmem/__main__.py (hunk 1 above)
Comment (ajparsons) [reply pyscraper/regmem/__main__.py:35]: Yep, made the same.

**Comment by ajparsons** [comment]
Added above - added some more commentary on the subitem movements because that wasn't clear. 

Added one extra fix that was needed for a march register to manually fetch a not present parent.

**Review by dracos — APPROVED** [review]
Thanks!

**Pull request merged**